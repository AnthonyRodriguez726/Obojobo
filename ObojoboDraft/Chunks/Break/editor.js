import React from 'react'

const BREAK_NODE = 'ObojoboDraft.Chunks.Break'

class Node extends React.Component {
	toggleSize() {
		const content = this.props.node.data.get('content')
		const newSize = (content.width === 'normal' ? 'large' : 'normal')
		content.width = newSize

		this.forceUpdate()
	}

	render() {
		const { isSelected } = this.props

		return (
			<div className={'component'}>
				<div
					className={`non-editable-chunk obojobo-draft--chunks--break viewer width-${this.props.node.data.get('content').width}`}
					{...this.props.attributes}>
					<hr/>
					{isSelected ? <button onClick={() => this.toggleSize()}>Toggle Size</button> : null}
				</div>
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock({
			type: BREAK_NODE,
			data: { content: { width: 'normal' } },
			isVoid: true
		})
		.collapseToStartOfNextText()
		.focus()
}

const slateToObo = node => {
	if (node.type !== BREAK_NODE) return null

	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content')

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case BREAK_NODE:
				return <Node {...props} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.Break': {
				isVoid: true,
			}
		}
	}
}

const Break ={
	components: {
		Node,
	},
	helpers: {
		insertNode,
		slateToObo,
	},
	plugins
}

export default Break
