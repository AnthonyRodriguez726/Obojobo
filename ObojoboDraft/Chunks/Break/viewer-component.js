import './viewer-component.scss'

import Common from 'Common'

let { OboComponent } = Common.components
let { NonEditableChunk } = Common.chunk

export default props => (
	<OboComponent model={props.model} moduleData={props.moduleData}>
		<NonEditableChunk
			className={`obojobo-draft--chunks--break viewer width-${props.model.modelState.width}`}
		>
			<hr />
		</NonEditableChunk>
	</OboComponent>
)
