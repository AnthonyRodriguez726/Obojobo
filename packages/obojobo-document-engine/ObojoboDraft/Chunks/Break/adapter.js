let Adapter = {
	toText(model) {
		return '---'
	}
}

export default Adapter
