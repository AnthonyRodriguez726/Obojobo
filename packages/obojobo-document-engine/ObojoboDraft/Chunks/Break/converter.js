const slateToObo = node => ({
	id: node.key,
	type: node.type,
	children: [],
	content: {
		width: node.data.get('content').width
	}
})

const oboToSlate = node => {
	const content = node.content
	if (!content.width) content.width = 'normal'

	return {
		object: 'block',
		key: node.id,
		type: node.type,
		data: {
			content
		}
	}
}

export default { slateToObo, oboToSlate }
