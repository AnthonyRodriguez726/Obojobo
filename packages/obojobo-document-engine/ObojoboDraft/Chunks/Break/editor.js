import React from 'react'

const BREAK_NODE = 'ObojoboDraft.Chunks.Break'

class Node extends React.Component {
	toggleSize() {
		const content = this.props.node.data.get('content')
		const newSize = content.width === 'normal' ? 'large' : 'normal'
		content.width = newSize

		this.forceUpdate()
	}

	render() {
		const { isSelected } = this.props

		return (
			<div className={'component'}>
				<div
					className={`non-editable-chunk obojobo-draft--chunks--break viewer width-${
						this.props.node.data.get('content').width
					}`}
					{...this.props.attributes}
				>
					<hr />
					{isSelected ? <button onClick={() => this.toggleSize()}>Toggle Size</button> : null}
				</div>
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock({
			type: BREAK_NODE,
			data: { content: { width: 'normal' } },
			isVoid: true
		})
		.collapseToStartOfNextText()
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = {
		width: node.data.get('content').width
	}
	json.children = []

	return json
}

const oboToSlate = node => {
	const json = {}
	json.object = 'block'
	json.key = node.id
	json.type = node.type
	json.isVoid = true
	json.data = { content: node.content }

	if (!json.data.content.width) json.data.content.width = 'normal'

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case BREAK_NODE:
				return <Node {...props} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.Break': {
				isVoid: true
			}
		}
	}
}

const Break = {
	components: {
		Node
	},
	helpers: {
		insertNode,
		slateToObo,
		oboToSlate
	},
	plugins
}

export default Break
