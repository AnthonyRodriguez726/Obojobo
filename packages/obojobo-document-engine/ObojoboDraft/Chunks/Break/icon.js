import React from 'react'

const Icon = () => {
	return (
		<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 18.17 16.52">
			<defs>
				<style>.cls-1{opacity:0.09;}.cls-2{fill:#231f20;}.cls-3{fill:none;stroke:#231f20;stroke-linecap:round;stroke-miterlimit:10;opacity:0.55;}</style>
			</defs>
			<title>break</title>
			<g class="cls-1">
				<rect class="cls-2" width="18.17" height="0.84"/>
				<rect class="cls-2" y="2.37" width="11.75" height="0.84"/>
				<rect class="cls-2" y="13" width="18.17" height="0.84"/>
				<rect class="cls-2" y="15.68" width="15.25" height="0.84"/>
			</g>
			<line class="cls-3" x1="3.44" y1="7.91" x2="14.73" y2="7.91"/>
		</svg>
	)
}

export default Icon
