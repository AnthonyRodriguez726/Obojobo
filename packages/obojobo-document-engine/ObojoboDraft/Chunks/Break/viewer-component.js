import './viewer-component.scss'

import React from 'react'

import Common from 'Common'
import Viewer from 'Viewer'

const { OboComponent } = Viewer.components
const { NonEditableChunk } = Common.chunk

const Break = props => (
	<OboComponent model={props.model} moduleData={props.moduleData}>
		<NonEditableChunk
			className={`obojobo-draft--chunks--break viewer width-${props.model.modelState.width}`}
		>
			<hr />
		</NonEditableChunk>
	</OboComponent>
)

export default Break
