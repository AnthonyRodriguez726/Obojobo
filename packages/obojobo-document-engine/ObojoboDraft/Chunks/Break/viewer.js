import adapter from './adapter'
import Common from 'Common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.Break', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
