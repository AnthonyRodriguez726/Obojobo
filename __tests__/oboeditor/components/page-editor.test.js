import { shallow } from 'enzyme'
import React from 'react'
import { Value } from 'slate'

jest.mock('../../../src/scripts/viewer/util/api-util')

import PageEditor from '../../../src/scripts/oboeditor/components/page-editor'
import Break from '../../../ObojoboDraft/Chunks/Break/editor'
import APIUtil from '../../../src/scripts/viewer/util/api-util'

const BOLD_MARK = 'b'
const ITALIC_MARK = 'i'
const STRIKE_MARK = 'del'
const QUOTE_MARK = 'q'
const LINK_MARK = 'a'

const CONTENT_NODE = 'ObojoboDraft.Sections.Content'
const ASSESSMENT_NODE = 'ObojoboDraft.Sections.Assessment'

describe('PageEditor', () => {
	test('EditorNav component', () => {
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with no page', () => {
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		component.setProps({ page: null })
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with no page updating to a page', () => {
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		component.setProps({ page: null })
		component.setProps(props)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with no page updating to a page', () => {
		const props = {
			page: {
				id: 1,
				set: jest.fn(),
				attributes: {
					children: []
				},
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		component.setProps({
			page: {
				id: 2,
				set: jest.fn(),
				attributes: {
					children: []
				},
				get: jest.fn()
			}
		})
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with content', () => {
		const props = {
			page: {
				id: 2,
				set: jest.fn(),
				attributes: {
					children: [
						{
							type: 'ObojoboDraft.Chunks.Break',
							content: {}
						},
						{
							type: 'NonsenseNode'
						}
					]
				},
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with content exports to database', () => {
		jest.spyOn(window, 'alert')
		window.alert.mockReturnValueOnce(null)
		APIUtil.postDraft.mockResolvedValueOnce({ status: 'ok' })

		const props = {
			page: {
				id: 2,
				set: jest.fn(),
				attributes: {
					children: [
						{
							type: 'ObojoboDraft.Chunks.Break',
							content: {}
						},
						{
							type: 'NonsenseNode'
						}
					]
				},
				get: jest
					.fn()
					.mockReturnValueOnce({
						// get('content') in
						scoreActions: [
							{
								for: '[0,100]',
								page: {
									type: 'ObojoboDraft.Pages.Page',
									children: []
								}
							}
						],
						rubric: {}
					})
					.mockReturnValueOnce(ASSESSMENT_NODE) // get('type') in import
					.mockReturnValueOnce({}) // get('content') in export
			},
			model: {
				children: [
					{
						get: () => ASSESSMENT_NODE
					},
					{
						get: () => CONTENT_NODE,
						flatJSON: () => {
							return { children: [] }
						},
						children: {
							models: [
								{
									get: () => null
								}
							]
						}
					},
					{
						get: () => 'mockNode'
					}
				],
				flatJSON: () => {
					return { children: [] }
				}
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		component
			.find('button')
			.at(14)
			.simulate('click')

		expect(tree).toMatchSnapshot()
		expect(APIUtil.postDraft).toHaveBeenCalle
	})

	test('EditorNav component with content exports to database with no rubric', () => {
		jest.spyOn(window, 'alert')
		window.alert.mockReturnValueOnce(null)
		APIUtil.postDraft.mockResolvedValueOnce({ status: 'ok' })

		const props = {
			page: {
				id: 2,
				set: jest.fn(),
				attributes: {
					children: [
						{
							type: 'ObojoboDraft.Chunks.Break',
							content: {}
						},
						{
							type: 'NonsenseNode'
						}
					]
				},
				get: jest
					.fn()
					.mockReturnValueOnce({
						// get('content') in
						scoreActions: [
							{
								for: '[0,100]',
								page: {
									type: 'ObojoboDraft.Pages.Page',
									children: []
								}
							}
						]
					})
					.mockReturnValueOnce(ASSESSMENT_NODE) // get('type') in import
					.mockReturnValueOnce({}) // get('content') in export
			},
			model: {
				children: [
					{
						get: () => ASSESSMENT_NODE
					},
					{
						get: () => CONTENT_NODE,
						flatJSON: () => {
							return { children: [] }
						},
						children: {
							models: [
								{
									get: () => null
								}
							]
						}
					},
					{
						get: () => 'mockNode'
					}
				],
				flatJSON: () => {
					return { children: [] }
				}
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		component
			.find('button')
			.at(14)
			.simulate('click')

		expect(tree).toMatchSnapshot()
		expect(APIUtil.postDraft).toHaveBeenCalle
	})

	test('EditorNav component with content fails to export to database', () => {
		jest.spyOn(window, 'alert')
		window.alert.mockReturnValueOnce(null)
		APIUtil.postDraft.mockResolvedValueOnce({ status: 'not ok' })

		const props = {
			page: {
				id: 2,
				set: jest.fn(),
				attributes: {
					children: [
						{
							type: 'ObojoboDraft.Chunks.Break',
							content: {}
						},
						{
							type: 'NonsenseNode'
						}
					]
				},
				get: jest.fn()
			},
			model: {
				children: [],
				flatJSON: () => {
					return { children: [] }
				}
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		component
			.find('button')
			.at(14)
			.simulate('click')

		expect(tree).toMatchSnapshot()
		expect(APIUtil.postDraft).toHaveBeenCalle
	})

	test('EditorNav component inserts item', () => {
		jest.spyOn(Break.helpers, 'insertNode')
		Break.helpers.insertNode.mockReturnValueOnce(null)
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		component
			.find('button')
			.at(2)
			.simulate('click')

		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component changes value', () => {
		jest.spyOn(Break.helpers, 'insertNode')
		Break.helpers.insertNode.mockReturnValueOnce(null)
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		component.find('.obojobo-draft--pages--page').simulate('change', { value: Value.create({}) })

		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component renders marks', () => {
		jest.spyOn(Break.helpers, 'insertNode')
		window.getSelection = jest.fn().mockReturnValueOnce({ rangeCount: 0 })
		Break.helpers.insertNode.mockReturnValueOnce(null)
		const props = {
			page: {
				attributes: { children: [] },
				get: jest.fn()
			}
		}
		const component = shallow(<PageEditor {...props} />)
		const tree = component.html()

		const editorProps = component.find('.obojobo-draft--pages--page').props()

		const mark = {
			children: 'mockChild',
			mark: { type: BOLD_MARK }
		}

		expect(editorProps.renderMark(mark)).toMatchSnapshot()
		mark.mark.type = ITALIC_MARK
		expect(editorProps.renderMark(mark)).toMatchSnapshot()
		mark.mark.type = STRIKE_MARK
		expect(editorProps.renderMark(mark)).toMatchSnapshot()
		mark.mark.type = QUOTE_MARK
		expect(editorProps.renderMark(mark)).toMatchSnapshot()
		mark.mark.type = LINK_MARK
		expect(editorProps.renderMark(mark)).toMatchSnapshot()

		expect(tree).toMatchSnapshot()
	})
})
