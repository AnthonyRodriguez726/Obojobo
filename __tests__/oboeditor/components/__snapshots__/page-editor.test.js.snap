// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageEditor EditorNav component 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component changes value 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component inserts item 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component toggles mark 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><!-- react-empty: 32 --><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component with content 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component with content exports to database 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component with content fails to export to database 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component with no page 1`] = `"<p>No content avalible, click on a page to start editing</p>"`;

exports[`PageEditor EditorNav component with no page updating to a page 1`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;

exports[`PageEditor EditorNav component with no page updating to a page 2`] = `"<div class=\\"editor\\"><div class=\\"toolbar\\"><div><button>Bold</button><button>Italic</button><button>Strikethrough</button><button>Quote</button><button>Monospace</button><button>Latex</button><button>Link</button><button>Superscript</button><button>Subscript</button><button>Left</button><button>Right</button><button>Center</button></div><div class=\\"dropdown\\"><button>+ Insert Node</button><div class=\\"drop-content\\"><button>ObojoboDraft.Chunks.ActionButton</button><button>ObojoboDraft.Chunks.Break</button><button>ObojoboDraft.Chunks.Code</button><button>ObojoboDraft.Chunks.Figure</button><button>ObojoboDraft.Chunks.Heading</button><button>ObojoboDraft.Chunks.IFrame</button><button>ObojoboDraft.Chunks.List</button><button>ObojoboDraft.Chunks.MathEquation</button><button>ObojoboDraft.Chunks.Table</button><button>ObojoboDraft.Chunks.Text</button><button>ObojoboDraft.Chunks.YouTube</button><button>ObojoboDraft.Chunks.QuestionBank</button><button>ObojoboDraft.Chunks.Question</button></div></div></div><button class=\\"exporter\\">Save Document</button></div>"`;
